text/0000-glimmer-components.md

```diff
@@ -0,0 +1,1019 @@
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Glimmer Components
+
+## Summary
+
+Glimmer components are a simpler, more ergonomic, and more declarative approach
+to building components. They represent the sum of multiple years of design and
+feature work by the community, which stemmed from the original RFCs and
+discussions surrounding "angle-bracket components".
+
+This RFC proposes adding Glimmer components to Ember's public API, and making
+them the default new app experience in Ember Octane.
+
+## First, a Bit of History
+
+As components became the standard for Single Page Apps (SPAs) several years
+ago, and the Ember community began adopting them in earnest and converting from
+Ember 1's primarily MVC oriented approach, there were many small and large
+issues that cropped up with Ember's component API and usage: `{{curly-bracket}}`
+syntax felt dated with the introduction of Web Components and other major
+frameworks; the inability to specify or override HTML attributes led to API
+explosions; the implicit wrapper element and customization fields (`tagName`,
+`classNames`, et al.) felt burdensome and made templates difficult to read
+compared to other frameworks; two way data-binding led to strange, hard to
+predict data cycles within apps; and so on.
+
+From this came the original [Angle Bracket component
+RFC](https://github.com/emberjs/rfcs/blob/0b32059b3704836e52c906a0ead64ac186c844d8/active/0000-component-unification.md).
+The idea was simple: switch to the superior `<angle-bracket>` syntax of web
+components, and solve all the other problems! Seems easy enough, right?
+
+As you can imagine, it was _not_ that easy. There was a flurry of discussion on
+the original RFC, and many ideas were thrown around. This was seen as the _one_
+chance Ember had to "fix" its component API, and the community did not want to
+get it wrong and lock us into yet _another_ set of painful papercuts. After much
+debate and lots of back and forth with the design, it was ultimately decided
+that attempting to redesign components all at once, monolithically, was too
+much. Instead, the individual ideas from that discussion could be broken out and
+implemented in isolation, in a backwards compatible way, both incrementally
+building a new, well thought out component API _and_ laying the groundwork in
+the framework for future redesigns.
+
+Many of the features from these discussions which are core to Glimmer
+components have already landed in Ember, including: Angle-bracket invocation,
+named arguments, element modifiers, and component managers.
+
+Glimmer components are the final piece of that puzzle. They include the last
+of the major features that were discussed during the original Angle Brackets
+RFC, and holistically those features make a much simpler and more ergonomic
+component API. Taken alone, they are an incremental change. Their individual
+features aren't _that_ much more than what we currently have in Ember today. But
```

Comment:
>  aren't _that_ much more than what we currently have in Ember today

that much more _what_? complex? performant?

Reply:
maybe "aren't _that_ revolutionary compared to what we have in Ember today"?